gatsby-config.js

```diff
@@ -131,22 +135,54 @@ const plugins = [
                   sort: { fields: [date], order: DESC }
                 ) {
                   nodes {
-                    html
+                    htmlAst
                     slug
                     title
                     date
                     description
                   }
                 }
               }
-            `,
+        `,
           serialize: ({ query: { site, allBlogPost } }) => {
+            /* This processor works on Rehype ASTs, doing two things:
+
+          1: All root-relative links are prepended with the site URL, making
+             them absolute. This uses data from siteMetadata, hence why it is
+             defined within this scope.
+
+          2: All images processed by Gatsby to be responsive are "unwrapped"
+             into their fallback 'img' nodes, as RSS doesn't work with the
+             tricks that true HTML does.
+           */
+            const processor = rehype()
+              .use(urls, url =>
+                url.href.startsWith('/')
+                  ? site.siteMetadata.site_url + url.href
+                  : undefined
+              )
+              .use(() => tree => {
+                // Unwrap all gatsby-resp-image-wrapper elements into plain images.
+                selectAll('.gatsby-resp-image-wrapper', tree).forEach(node => {
+                  const fallbackImg = select('img', node)
+                  delete node.children
+                  Object.assign(node, fallbackImg)
+                  node.properties = {
+                    ...node.properties,
+                    style: 'max-width: 100%'
+                  }
+                })
+                return tree
+              })
+              .freeze()
+
             return allBlogPost.nodes.map(node => {
+              const html = processor.stringify(processor.runSync(node.htmlAst))
```

Comment:
check the `const { convertHtmlToHast, convertHastToHtml } = require('../utils/convertHast')` ?

Reply:
That looks like a good place to extract this function. My parser uses different plugins so I can't just use that one as it is nor change it to mine, but it can sit next to the other rehype plugins.

Reply:
I started using the convertHastToHtml for the latter half of the feed processor, after the AST is transformed.